src/util.cpp

```diff
@@ -536,6 +565,58 @@ void ArgsManager::ForceSetArg(const std::string& strArg, const std::string& strV
     m_override_args[strArg] = {strValue};
 }
 
+void ArgsManager::AddArg(const std::string& name, const std::string& help, const bool debug_only, const OptionsCategory& cat)
+{
+    std::pair<OptionsCategory, std::string> key(cat, name);
+    assert(m_available_args.count(key) == 0);
```

Comment:
nit: I guess the key can just be the name, as two names with a different category don't make sense?

Reply:
It's a hack to make std::map sort the keys by category and then alphabetically when iterating over the map so that the help text comes out correctly.

Reply:
I doubt it matters in practice, so I consider it a code style nit.